src/community/geogig/src/main/java/org/geogig/geoserver/config/RepositoryInfo.java

```diff
@@ -84,14 +92,29 @@ void setName(String name) {
         this.name = name;
     }
 
+    public void setRepoName(String name) {
+        this.repoName = name;
+    }
+
     public String getRepoName() {
-        if (this.location != null)  {
-        	try {
-        		Repository repo = RepositoryResolver.load(this.location);
-        		return repo.command(ResolveRepositoryName.class).call();
-        	} catch (RepositoryConnectionException e) {
-        		Throwables.propagate(e);
-        	}
+        if (this.repoName != null) {
+            return repoName;
+        }
+        if (this.location != null) {
+            try {
+                // lookup the resolver
+                RepositoryResolver resolver = RepositoryResolver.lookup(this.location);
+                // if the repo exists, get the name from it
+                if (resolver.repoExists(this.location)) {
+                    // it exists, load it and fetch the name
+                    Repository repo = RepositoryResolver.load(this.location);
```

Comment:
looks like a `repo.close()` wouldn't hurt after this. Actually would be a desirable thing